src/realm/geospatial.cpp

```diff
@@ -109,17 +109,13 @@ Geospatial Geospatial::from_obj(const Obj& obj, ColKey type_col, ColKey coords_c
     }
 
     Lst<double> coords = obj.get_list<double>(coords_col);
-    GeoPoint geo;
-    if (coords.size() >= 1) {
-        geo.longitude = coords[0];
-    }
-    if (coords.size() >= 2) {
-        geo.latitude = coords[1];
-    }
-    if (coords.size() >= 3) {
-        geo.set_altitude(coords[2]);
-    }
-    return geo;
+    if (coords.size() < 2)
+        throw SerializationError("Invalid number of coordinates for geospatial geometry");
```

Comment:
I don't think it is a SerializationError. Actually, I think we should be returning an invalid Geospatial in this case, because we cannot enforce data to conform to a particular format given that it can be modified by the server, or by manipulating the embedded object manually. (here and in from_link below)